workspace/graph-sequencer/test/index.ts

```diff
@@ -0,0 +1,366 @@
+import { graphSequencer } from '../src'
+
+test('base test case0', () => {
+  expect(graphSequencer(new Map([
+    ['a', ['a']],
+    ['b', ['b']],
+    ['c', ['c']],
+  ]
+  ), ['a', 'b', 'c'])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [['a', 'b', 'c']],
+      cycles: [
+        ['a'], ['b'], ['c'],
+      ],
+    }
+  )
+})
+
+test('base test case1', () => {
+  expect(graphSequencer(new Map([
+    ['a', ['b', 'c']],
+    ['b', ['b']],
+    ['c', ['b', 'c']]]
+  ), ['a', 'b', 'c'])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [['b', 'c'], ['a']],
+      cycles: [
+        ['b'], ['c'],
+      ],
+    }
+  )
+})
+
+test('base test case2', () => {
+  expect(graphSequencer(new Map([
+    ['a', ['b', 'c']],
+    ['b', []],
+    ['c', ['b']]]
+  ), ['a', 'b', 'c'])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [['b'], ['c'], ['a']],
+      cycles: [],
+    }
+  )
+})
+
+test('base test case3', () => {
+  expect(graphSequencer(new Map([
+    ['a', ['b', 'c']],
+    ['b', []],
+    ['c', []],
+    ['d', ['a']],
+    ['e', ['a', 'b', 'c']]]
+  ), ['a', 'd', 'e'])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [['a'], ['d', 'e']],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with no dependencies', () => {
+  expect(graphSequencer(new Map([
+    [0, []],
+    [1, []],
+    [2, []],
+    [3, []],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [[0, 1, 2, 3]],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with no dependencies sub graph', () => {
+  expect(graphSequencer(new Map([
+    [0, []],
+    [1, []],
+    [2, []],
+    [3, []],
+  ]), [0, 1, 2])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [[0, 1, 2]],
+      cycles: [],
+    }
+  )
+})
+
+test('graph with multiple dependencies on one item', () => {
+  expect(graphSequencer(new Map([
+    [0, [3]],
+    [1, [3]],
+    [2, []],
+    [3, []],
+  ]), [0, 1, 2, 3])).toStrictEqual(
+    {
+      safe: true,
+      chunks: [[2, 3], [0, 1]],
+      cycles: [],
+    }
+  )
+})
+
```

Comment:
This is not related to your changes but I guess our script runners are not as affective as they should be at the moment. 

We don't really need to wait for both 2 and 3 to finish before running 0 an 1. We only need to wait for 3 to finish and then we can continue with 0 and 1 while 2 might be still in progress.

Reply:
The behavior is the same as the old version

```
graphSequencer({
  graph: new Map([
    ['a', ['c']],
    ['b', ['c']],
    ['c', []],
    ['d', []],
  ]),
  groups: [['a', 'b', 'c','d']]
})
// { safe: true, chunks: [ [ 'c', 'd' ], [ 'a', 'b' ] ], cycles: [] }

```